CRM/Civicase/Service/CaseTypeCategoryEventHandler.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+/**
+ * Class CRM_Civicase_Service_CaseTypeCategoryEventHandler.
+ */
+class CRM_Civicase_Service_CaseTypeCategoryEventHandler {
+
+  /**
+   * Menu handler.
+   *
+   * @var \CRM_Civicase_Service_CaseCategoryMenu
+   */
+  protected $menu;
+
+  /**
+   * Custom data handler.
+   *
+   * @var \CRM_Civicase_Service_CaseCategoryCustomDataType
+   */
+  protected $customData;
+
+  /**
+   * Custom field handler.
+   *
+   * @var \CRM_Civicase_Service_CaseCategoryCustomFieldExtends
+   */
+  protected $customFieldExtends;
+
+  /**
+   * Should we check referer before running action?
+   *
+   * @var bool
+   */
+  protected $needCheckReferer = FALSE;
+
+  /**
+   * Should we check option group before running action?
+   *
+   * Depends on $optionGroupId.
+   *
+   * @var bool
+   */
+  protected $needCheckOptionGroup = TRUE;
+
+  /**
+   * Case type category option group id. Used to check option group.
+   *
+   * @var int
+   */
+  protected $optionGroupId;
+
+  /**
+   * Object name.
+   *
+   * Is used to check object if it's type is unpredictable or unknown.
+   *
+   * @var string
+   */
+  protected $objectName = 'OptionValue';
+
+  /**
+   * CRM_Civicase_Service_CaseTypeCategoryEventHandler constructor.
+   *
+   * @param object $menu
+   *   Menu handler.
+   * @param object $customData
+   *   Custom data handler.
+   * @param object $customFieldExtends
+   *   Custom field handler.
+   */
+  public function __construct($menu, $customData, $customFieldExtends) {
+    $this->menu = $menu;
+    $this->customData = $customData;
+    $this->customFieldExtends = $customFieldExtends;
+  }
+
+  /**
+   * Perform actions on case type category create.
+   *
+   * @param string $name
```

Comment:
Can we rename this to `caseCategoryName` ?

Reply:
All this class is about case type categories. So why should we add `caseCategory` prefix to all variables? Why not follow 'Write less - Do more' approach in this case?

Reply:
If it were a function, I would not have suggested this. But being a variable, it's easier to read `CaseCategoryName` than just `Name`. It's clearer. Name could be anything.